Extension/src/main.ts

```diff
@@ -11,331 +11,150 @@ import * as util from './common';
+    tempCommandRegistrar = new commands.TemporaryCommandRegistrar();
     util.setExtensionContext(context);
     Telemetry.activate();
     util.setProgress(0);
+    cpptoolsJsonUtils.activate(context);
+    initializeInstallBlob();
 
     // Initialize the DebuggerExtension and register the related commands and providers.
     DebuggerExtension.initialize();
 
-    if (context.globalState.get<number>(userBucketString, -1) == -1) {
-        let bucket: number = Math.floor(Math.random() * userBucketMax) + 1; // Range is [1, userBucketMax].
-        context.globalState.update(userBucketString, bucket);
-    }
-
-    // Add temp commands that invoke the real commands after download/install is complete (preventing an error message),
-    // and also show the C/C++ output pane and the "wait for download" message.
-    tempCommands = [];
-    delayedCommandsToExecute = new Set<string>();
-    registerTempCommand("C_Cpp.ConfigurationEdit");
-    registerTempCommand("C_Cpp.ConfigurationSelect");
-    registerTempCommand("C_Cpp.SwitchHeaderSource");
-    registerTempCommand("C_Cpp.Navigate");
-    registerTempCommand("C_Cpp.GoToDeclaration");
-    registerTempCommand("C_Cpp.PeekDeclaration");
-    registerTempCommand("C_Cpp.ToggleErrorSquiggles");
-    registerTempCommand("C_Cpp.ToggleIncludeFallback");
-    registerTempCommand("C_Cpp.ShowReleaseNotes");
-    registerTempCommand("C_Cpp.ResetDatabase");
-    registerTempCommand("C_Cpp.PauseParsing");
-    registerTempCommand("C_Cpp.ResumeParsing");
-    registerTempCommand("C_Cpp.ShowParsingCommands");
-    registerTempCommand("C_Cpp.TakeSurvey");
-
-    processRuntimeDependencies(() => {
-        util.readFileText(util.getExtensionFilePath("cpptools.json"))
-            .then((cpptoolsString) => {
-                processCpptoolsJson(cpptoolsString);
-            })
-            .catch((error) => {
-                // We already log telemetry if cpptools.json fails to download.
-            })
-            .then(() => {
-                // Main activation code.
-                tempCommands.forEach((command) => {
-                    command.dispose();
-                });
-                tempCommands = [];
-                LanguageServer.activate(delayedCommandsToExecute);
-                delayedCommandsToExecute.forEach((command) => {
-                    vscode.commands.executeCommand(command);
-                });
-                delayedCommandsToExecute.clear();
-            })
-            .then(() => {
-                // Redownload cpptools.json after activation so it's not blocked.
-                // It'll be used after the extension reloads.
-                downloadCpptoolsJsonPkg();
-            });
-    });
-
-    setInterval(() => {
-        // Redownload occasionally to prevent an extra reload during long sessions.
-        downloadCpptoolsJsonPkg();
-    }, 30 * 60 * 1000); // 30 minutes.
+    return processRuntimeDependencies();
```

Comment:
>processRuntimeDependencies [](start = 11, length = 26)

should this be awaited?

Reply:
I don't think we want vscode to hang on this activation?